sound/soc/sof/intel/hda-loader.c

```diff
@@ -182,11 +182,14 @@ static int cl_dsp_init(struct snd_sof_dev *sdev, int stream_tag)
 err:
 	flags = SOF_DBG_DUMP_REGS | SOF_DBG_DUMP_PCI | SOF_DBG_DUMP_MBOX;
 
-	/* force error log level after max boot attempts */
+	/*
+	 * after max boot attempts make sure that the dump is printed and error
+	 * log level is used
+	 */
 	if (hda->boot_iteration == HDA_FW_BOOT_ATTEMPTS)
-		flags |= SOF_DBG_DUMP_FORCE_ERR_LEVEL;
+		flags |= SOF_DBG_DUMP_FORCE_ERR_LEVEL | SOF_DBG_DUMP_FORCE;
```

Comment:
why is SOF_DBG_DUMP_FORCE needed here but not when cl_copy_fw() fails?

Reply:
Here the final dump need to be forced as we printed for the first one. cl_copy_fw() will only happen if the boot is successful.

But this got me thinking and I think I got this wrong... Instead of `SOF_DBG_DUMP_FORCE` we should have `SOF_DBG_DUMP_OPTIONAL` and it should be set for the intermediate dumps to not print them at all (they will be if the module param flag to print dumps is set).

With the current implementation if we need two attempts to boot then the DSP dump will be marked as printed and if a panic happens later we will not print the info we would need to debug.

Let me try to get this right tomorrow.